reconcile/utils/oc.py

```diff
@@ -575,22 +579,36 @@ def recycle_pods(self, dry_run, namespace, dep_kind, dep_resource):
 
         for kind, objs in recyclables.items():
             for obj in objs:
-                name = obj['metadata']['name']
-                logging.info([f'recycle_{kind.lower()}',
-                              self.cluster_name, namespace, name])
-                if not dry_run:
-                    now = datetime.now()
-                    recycle_time = now.strftime("%d/%m/%Y %H:%M:%S")
-
-                    # honor update strategy by setting annotations to force
-                    # a new rollout
-                    a = obj['spec']['template']['metadata'].get(
-                        'annotations', {})
-                    a['recycle.time'] = recycle_time
-                    obj['spec']['template']['metadata']['annotations'] = a
-                    cmd = ['apply', '-n', namespace, '-f', '-']
-                    stdin = json.dumps(obj, sort_keys=True)
-                    self._run(cmd, stdin=stdin, apply=True)
+                self.recycle(dry_run, namespace, kind, obj)
+
+    @retry(exceptions=ObjectHasBeenModifiedError)
+    def recycle(self, dry_run, namespace, kind, obj):
+        """Recycles an object by adding a recycle.time annotation
+
+        Args:
+            dry_run (bool): Is this a dry run
+            namespace (string): Namespace to work in
+            kind (string): Object kind
+            obj (dict): Object to recycle
+        """
```

Comment:
NIT: (And not sure if we agreed that we can use both google and sphinx) But the docstring is not following the guidelines on the dev guide https://service.pages.redhat.com/dev-guidelines/docs/appsre/python-style-guide/#docstrings

Reply:
Besides of that lgtm